lib/trino-hive-formats/src/test/java/io/trino/hive/formats/avro/TestAvroBase.java

```diff
@@ -0,0 +1,457 @@
+import static java.lang.Double.doubleToLongBits;
+import static java.lang.Float.floatToIntBits;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.within;
+
+public abstract class TestAvroBase
+{
+    protected static final TypeOperators TYPE_OPERATORS = new TypeOperators();
+    protected static final ArrayType ARRAY_INTEGER = new ArrayType(INTEGER);
+    protected static final MapType MAP_VARCHAR_VARCHAR = new MapType(VARCHAR, VARCHAR, TYPE_OPERATORS);
+    protected static final MapType MAP_VARCHAR_INTEGER = new MapType(VARCHAR, INTEGER, TYPE_OPERATORS);
+    static final TrinoFileSystem TRINO_LOCAL_FILESYSTEM = new LocalFileSystem(Path.of("/"));
+    static final Path tempDirectory;
+
+    protected static final Schema SIMPLE_RECORD_SCHEMA = SchemaBuilder.record("simpleRecord")
+            .fields()
+            .name("a")
+            .type().intType().noDefault()
+            .name("b")
+            .type().doubleType().noDefault()
+            .name("c")
+            .type().stringType().noDefault()
+            .endRecord();
+
+    protected static final Schema SIMPLE_ENUM_SCHEMA = SchemaBuilder.enumeration("myEnumType").symbols("A", "B", "C");
+
+    protected static final Schema ALL_TYPES_RECORD_SCHEMA = SchemaBuilder.builder()
+            .record("all")
+            .fields()
+            .name("aBoolean")
+            .type().booleanType().noDefault()
+            .name("aInt")
+            .type().intType().noDefault()
+            .name("aLong")
+            .type().longType().noDefault()
+            .name("aFloat")
+            .type().floatType().noDefault()
+            .name("aDouble")
+            .type().doubleType().noDefault()
+            .name("aString")
+            .type().stringType().noDefault()
+            .name("aBytes")
+            .type().bytesType().noDefault()
+            .name("aFixed")
+            .type().fixed("myFixedType").size(16).noDefault()
+            .name("anArray")
+            .type().array().items().intType().noDefault()
+            .name("aMap")
+            .type().map().values().intType().noDefault()
+            .name("anEnum")
+            .type(SIMPLE_ENUM_SCHEMA).noDefault()
+            .name("aRecord")
+            .type(SIMPLE_RECORD_SCHEMA).noDefault()
+            .name("aUnion")
+            .type().optional().stringType()
+            .endRecord();
+    protected static final GenericRecord ALL_TYPES_GENERIC_RECORD;
+    protected static final Page ALL_TYPES_PAGE;
+    protected static final GenericRecord SIMPLE_GENERIC_RECORD;
+    protected static final String A_STRING_VALUE = "a test string";
+    protected static final ByteBuffer A_BYTES_VALUE = ByteBuffer.wrap("a test byte array".getBytes(StandardCharsets.UTF_8));
+    protected static final GenericData.Fixed A_FIXED_VALUE;
+
+    static {
+        try {
+            tempDirectory = Files.createTempDirectory("testingAvro");
```

Comment:
This needs to be cleaned up. Move it into a `@BeforeClass` and delete it in `@AfterClass(alwaysRun = true)`